libsolidity/formal/Z3CHCInterface.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libsolidity/formal/Z3CHCInterface.h>
+
+#include <liblangutil/Exceptions.h>
+#include <libdevcore/CommonIO.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::solidity::smt;
+
+Z3CHCInterface::Z3CHCInterface():
+	m_solver(m_context),
+	m_variables(m_context)
+{
+	// This needs to be set globally.
+	z3::set_param("rewriter.pull_cheap_ite", true);
```

Comment:
Actually we can remove this.

Reply:
It's important for the SMTChecker because it relies heavily on `ite` to join SSA variables from branches (phi function), but here it shouldn't be needed.